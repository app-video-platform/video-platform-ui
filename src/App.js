import { Routes, Route } from "react-router-dom";

import Navigation from "./pages/nav/nav.component";
import About from "./pages/about/about.component";
import Home from "./pages/home/home.component";
import Contact from "./pages/contact/contact.component";

import "./App.css";

function App() {
	return (
    <div>
		<Routes>
			<Route path="/" element={<Navigation />}>
				<Route index element={<Home />} />
				<Route path="about" element={<About />} />
				<Route path="contact" element={<Contact />} />
			</Route>
		</Routes>
    </div>
	);
}

export default App;
