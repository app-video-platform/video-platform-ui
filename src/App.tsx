import React from 'react';
import { Routes, Route } from 'react-router-dom';
import Home from './pages/home/home.component';
import About from './pages/about/about.component';
import EmailSent from './pages/email-sent/email-sent.component';
import SignIn from './pages/sign-in/sign-in.component';
import SignUp from './pages/sign-up/sign-up.component';
import VerifyEmail from './pages/verify-email/verify-email.component';
import AppInitializer from './utils/app-initializer.util';
import Navigation from './pages/nav/nav.component';
import Contact from './pages/contact/contact.component';
import NotFoundPage from './pages/errors/not-found/not-found.component';
import UnauthorizedPage from './pages/errors/unauthorized/unauthorized.component';
import Dashboard from './galactica-app/dashboard/dashboard.component';
import UserDashboard from './galactica-app/user-dashboard/user-dashboard.component';
import ProtectedRoute from './utils/protected-route.util';

const App = () => (
  <AppInitializer>
    <Routes>
      <Route path="/" element={<Navigation />}>
        <Route index element={<Home />} />
        <Route path="about" element={<About />} />
        <Route path="contact" element={<Contact />} />
      </Route>
      <Route path="signup" element={<SignUp />} />
      <Route path="signin" element={<SignIn />} />
      <Route path="/verify-email" element={<VerifyEmail />} />
      <Route path="/email-sent" element={<EmailSent />} />
      <Route path="*" element={<NotFoundPage />} />
      <Route path="/unauthorized" element={<UnauthorizedPage />} />

      <Route
        path="dashboard"
        element={
          // <ProtectedRoute>
          <Dashboard />
          // </ProtectedRoute>
        }
      />
      {/* <Route path='' element={<UserDashboard />} /> */}
    </Routes>

  </AppInitializer>
);

export default App;