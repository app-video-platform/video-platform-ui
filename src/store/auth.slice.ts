import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
import { User } from '../models/user';
import { registerUser } from '../api/user-api';
import { SignupFormData } from '../pages/sign-up/sign-up.component';

interface AuthState {
  user: null | User;
  token: string | null;
  loading: boolean;
  error: string | null;
}

const initialState: AuthState = { user: null, token: null, loading: false, error: null };

export const signupUser = createAsyncThunk<
  { user: User; token: string }, // Return type
  SignupFormData, // Argument type (user data)
  { rejectValue: string } // Error type
>('auth/signupUser', async (userData, { rejectWithValue }) => {
  try {
    const response = await registerUser(userData);
    return { user: response, token: 'mocked-jwt-token' }; // Assume API returns a token
  } catch (error: any) {
    return rejectWithValue(error.response?.data?.message || 'Signup failed');
  }
});


// ** Auth Slice **
const authSlice = createSlice({
  name: 'auth',
  initialState,
  reducers: {
    logout: (state) => {
      state.user = null;
      state.token = null;
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(signupUser.pending, (state) => {
        state.loading = true;
        state.error = null;
      })
      .addCase(signupUser.fulfilled, (state, action: PayloadAction<{ user: User; token: string }>) => {
        state.loading = false;
        state.user = action.payload.user;
        state.token = action.payload.token;
      })
      .addCase(signupUser.rejected, (state, action) => {
        state.loading = false;
        state.error = action.payload || 'Signup failed';
      });
  },
});

export const { logout } = authSlice.actions;
export default authSlice.reducer;
