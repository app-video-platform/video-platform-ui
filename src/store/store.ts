import { configureStore, createListenerMiddleware } from '@reduxjs/toolkit';
import authReducer from './auth-store/auth.slice';
import productsReducer from './product-store/product.slice';
import notificationsReducer from './notifications/notifications.slice';
import { setupProductListeners } from './listeners/product-listeners';

const listenerMiddleware = createListenerMiddleware();

setupProductListeners(listenerMiddleware);

export const store = configureStore({
  reducer: {
    auth: authReducer,
    products: productsReducer,
    notifications: notificationsReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().prepend(listenerMiddleware.middleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
