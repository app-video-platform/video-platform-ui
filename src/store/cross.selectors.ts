import { RootState } from './store';

export const makeSelectWishlistToCartState = (id: string) => (s: RootState) => {
  const inWishlist = s.wishlist.products.some((p) => p.id === id);
  const inCart = s.shopCart.products.some((p) => p.id === id);
  return { inWishlist, inCart };
};
