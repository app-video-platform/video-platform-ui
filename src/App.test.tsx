// App.test.tsx

import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from './App';
import '@testing-library/jest-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mockInitialState } from './utils/store.mock';

// Optional: include jest-dom matchers if not already globally set up
// import '@testing-library/jest-dom/extend-expect';

const mockStore = configureStore([]);
const store = mockStore(mockInitialState); // pass initial state as needed

describe('App Component', () => {
  test('renders without crashing', () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>
    );

    // Verify that the Navigation component is rendered.
    // Adjust the query if Navigation doesn't contain text "Navigation".
    // For now, we assume Navigation renders a nav element.
    expect(screen.getByRole('navigation')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>

      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
