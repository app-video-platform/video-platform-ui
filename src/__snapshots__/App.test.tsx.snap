// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<div>
  <nav>
    <div
      class="logo-container"
    />
    <div
      class="nav-links"
    >
      <ul
        class="nav-link-list"
      >
        <li>
          Home
        </li>
        <li>
          Chi sonno
        </li>
        <li>
          Contattami
        </li>
      </ul>
    </div>
    <div
      class="user-buttons"
    >
      <ul
        class="nav-link-list"
      >
        <ul
          class="nav-link-list"
        >
          <li>
            <button
              class="vp-button vp-button--secondary"
            >
              Sign Up
            </button>
          </li>
          <li>
            <button
              class="vp-button vp-button--primary"
            >
              Sign In
            </button>
          </li>
        </ul>
      </ul>
    </div>
  </nav>
  <div>
    Hoe wors
    <button
      class="bg-blue-500 text-white p-2 rounded hover:bg-blue-600 transition"
    >
      Fetch Data
    </button>
  </div>
</div>
`;
